node/caching-compiler.js

```diff
@@ -103,16 +111,18 @@ function readFileOrNull(filename) {
   }
 }
 
-function getCacheFilename(content, reify) {
-  const hash = createHash("sha1")
-    .update(reifyVersion).update("\0")
-    .update(content).update("\0");
-
-  if (reify) {
-    hash.update(JSON.stringify(reify));
+function getCacheFilename() {
+  const args = [];
+  const argc = arguments.length;
+  for (let i = 0; i < argc; ++i) {
+    args.push(arguments[i]);
   }
 
-  return hash.digest("hex") + ".js";
+  return createHash("sha1")
+    .update(reifyVersion)
+    .update("\0")
+    .update(JSON.stringify(args))
```

Comment:
Maybe just `JSON.stringify` on `reify` when it exists since when it does it will be a valid JSON object.